src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/PrisonApiClient.kt

```diff
@@ -5,20 +5,58 @@ import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.core.ParameterizedTypeReference
 import org.springframework.stereotype.Service
 import org.springframework.web.reactive.function.client.WebClient
+import uk.gov.justice.digital.hmpps.manageoffencesapi.model.PrisonApiHoCode
 import uk.gov.justice.digital.hmpps.manageoffencesapi.model.PrisonApiOffence
+import uk.gov.justice.digital.hmpps.manageoffencesapi.model.PrisonApiStatute
 import uk.gov.justice.digital.hmpps.manageoffencesapi.model.RestResponsePage
 
 @Service
 class PrisonApiClient(@Qualifier("prisonApiWebClient") private val webClient: WebClient) {
   private inline fun <reified T> typeReference() = object : ParameterizedTypeReference<T>() {}
   private val log = LoggerFactory.getLogger(this::class.java)
 
-  fun getOffences(pageNumber: Int): RestResponsePage<PrisonApiOffence> {
+  fun findByOffenceCodeStartsWith(offenceCode: String, pageNumber: Int): RestResponsePage<PrisonApiOffence> {
     log.info("Fetching all offences from prison-api for page number $pageNumber")
     return webClient.get()
-      .uri("/api/offences/all?page=$pageNumber&size=1000&sort=code,ASC")
+      .uri("/api/offences/code/$offenceCode?page=$pageNumber&size=1000&sort=code,ASC")
       .retrieve()
       .bodyToMono(typeReference<RestResponsePage<PrisonApiOffence>>())
       .block()!!
   }
+
+  fun createHomeOfficeCode(prisonApiHoCode: PrisonApiHoCode) {
+    webClient.post()
+      .uri("/api/offences/ho-code")
+      .bodyValue(prisonApiHoCode)
+      .retrieve()
+      .toBodilessEntity()
+      .block()
+  }
```

Comment:
I guess this is going to be doing a lot of updates... I was just wondering what would happen if one of these updates fails for some reason? Would it be simple to recover?